### PR TITLE
Docker

### PR DESCRIPTION
**Pull request by ajshedivy**
The goal is to "dockerize" pathlinker. Follow the instructions in the readme to build the pathlinker image

### CODE REVIEW

File: docker/README.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# Dockerized PathLinker
+
+- Use `COPY` statements to make container more dynamic
+
```
Comment (agitter) [inline docker/README.md]:
```suggestion
```
This may have been left over from an earlier version and can be deleted.

**Review by agitter — CHANGES_REQUESTED** [review]
@ajshedivy and I have been discussing different versions of the Dockerfile, and this version looked good to me.  There is one residual comment in the readme that can be deleted.

We also experimented with Docker operations in GitHub Actions continuous integration.  By adding a [workflow file](https://github.com/agitter/PathLinker/blob/1420c7f95786d88a5106d2bc2abb857deb8583ca/.github/workflows/docker.yaml), GitHub can automatically [build](https://github.com/agitter/PathLinker/runs/974090584#step:4:1) the Docker image every time the repository is updated.  It can also automatically tag the image and deploy it to DockerHub.  That isn't necessary now but could be considered in the future.

**Comment by ajshedivy** [comment]
RE: running the docker container on the example data.

In order to run the container:

first make sure you are in the Pathlinker root directory on your local machine

To run in Git for Windows and mount the example data inside the container:

`winpty docker run -v /$(pwd)/example:/home/PathLinker/example -it pathlinker/pathlinker bash`

then inside the container in the /home/Pathlinker/example directory call:

`python ../run.py sample-in-net.txt sample-in-nodetypes.txt`


The first command starts the container and opens up an interactive bash. Once the docker container is running move into the example directory (`/home/PathLinker/example`) and then call the second command to produce the `outk-100-ranked-edges.txt` file. This file should show up in the local example directory within the Pathlinker repository  since with used the `-v` flag.

File: docker/README.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
+# Dockerized PathLinker
+
+## Steps
+- Build Pathlinker docker container
```
Comment (agitter) [inline docker/README.md]: May want to keep the capitalization consistent in these lines: `PathLinker`

File: docker/README.md [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+## Run (Windows)
+
+To use the docker container with windows, use a bash terminal such as Git for Windows.
+
+To run container in Git for Windows:
+
+`winpty docker run -it pathlinker/pathlinker bash`
+
+**Note the** `winpty` **cammand**
```
Comment (agitter) [inline docker/README.md]:
```suggestion
**Note the** `winpty` **command**
```

**Pull request merged**